vm-setup/roles/v1aX_integration_test/tasks/remediation.yml

```diff
@@ -0,0 +1,127 @@
+---
+  - name: Get the name of the worker node
+    shell: |
+       kubectl get bmh -n "{{ NAMESPACE }}" -o json | jq -r '.items[]
+       | select(.spec.consumerRef.name | contains("{{ CLUSTER_NAME }}-workers"))
+       | .metadata.name'
+    register: worker_node_name
+
+  - name: Define WORKER_NODE variable with worker node name
+    set_fact:
+      WORKER_NODE: "{{ worker_node_name.stdout }}"
+
+  - name: Get the name of the worker libvirt VM
+    set_fact:
+      WORKER_NODE_VM: "node_0"
+    when: WORKER_NODE == "node-0"
+
+  - name: Get the name of the worker libvirt VM
+    set_fact:
+      WORKER_NODE_VM: "node_1"
+    when: WORKER_NODE == "node-1"
+
+  - name: Get the name of the worker libvirt VM
+    set_fact:
+      WORKER_NODE_VM: "node_2"
+    when: WORKER_NODE == "node-2"
+
+  - name: Get the name of the worker libvirt VM
+    set_fact:
+      WORKER_NODE_VM: "node_3"
+    when: WORKER_NODE == "node-3"
+
+  - name: Reboot "{{ WORKER_NODE }}"
+    shell: |
+       kubectl annotate bmh "{{ WORKER_NODE }}" -n "{{ NAMESPACE }}" reboot.metal3.io=
+
+  - name: List only shutdown VMs
+    virt:
+      command: list_vms
+      state: shutdown
+    register: shutdown_vms
+    retries: 50
+    delay: 10
+    until: shutdown_vms.list_vms == ["{{ WORKER_NODE_VM }}"]
```

Comment:
what if the other nodes are off too ? this might not match, because you would have three nodes shutdown

Reply:
That's a good point. I had the same issue for checking running VMs and that's why I didn't include it.

Reply:
Should we then exclude libvirt checking? or add a small shell script for checking node name in the array of strings ?

Reply:
what about until: 'WORKER_NODE_VM in shutdown_vms.list_vms' or something of the same idea. you can use the in operator. you could also use the select filter, to match that vm name only. There are quite some other alternatives

Reply:
Thanks, done.